src/ape/contracts/base.py

```diff
@@ -559,7 +611,88 @@ def poll_logs(
             yield from self.range(new_block.number, stop=new_block.number + 1)
 
 
-class ContractInstance(BaseAddress):
+class ContractTypeWrapper(ManagerAccessMixin):
+    contract_type: ContractType
+
+    def encode_input(self, *args: Any) -> HexBytes:
```

Comment:
I am wondering if we should drop this method. It seems weird to rely on number of args. Brownie had this so I included it but I can't imagine wanting to rely on this. Thoughts? Maybe I am misunderstanding something as well.

Better if you know at least the method name so you can do `contract.theMethod.encode_input(arg0, arg1, ...)`

Reply:
yeah, I do agree this would be weird to use if there were matching input lengths and types e.g. `ERC20.transfer` and `ERC20.approve`

Reply:
- [ ] Delete this method.